doc/ml_features.rst

```diff
@@ -0,0 +1,337 @@
+Instead of Neural Networks, or, in general, Machine Learning models, it is
+possible to upload to the DB (collections of) functions which can be used e.g.
+to perform pre- or post-processing operations on tensors stored on the DB.
+
+Since the functions are going to be stored as TorchScript modules, they
+- need to be jit-traceable
+- can use ``torch`` as a built-in module
+- can **not** import modules
+
+The easiest way of defining and storing functions on the DB is to create a
+dedicated file. In that file, we can define functions which will be callable
+through the SmartRedis ``Client``, but also from other functions in the
+same file. A typical script file would look like this:
+
+.. code-block:: python
+
+    def rescale(tensor, mu: float, sigma: float):
+        mean = tensor.mean()
+        std = tensor.std()
+
+        normalized = (tensor-mean)/std
+        return tensor*sigma + mu
+
+    def shift_y_to_x(x, y):
+        mu_x = x.mean()
+        sigma_x = x.std()
+        y_rescaled = rescale(y, mu_x, sigma_x)
+
+        return y_rescaled
+
+In the script, we defined ``shift_y_to_x``,
+a function which returns a modified copy of a tensor ``y``,
+which matches the statistical distribution of the tensor ``x``.
+Notice that we are not importing ``torch`` in the script, as it will
+be recognized as a built-in by the TorchScript compiler. Because
+of the discrepancy between TorchScript's and Python's syntaxes, TorchScript
+scripts cannot be run as standalone Python scripts.
+
+Here is the code which allows us to run the function ``shift_y_to_x`` on
+tensors stored in the DB. We will assume that the above script is stored
+as ``"./shift.script"``.
+
+.. code-block:: python
+
+    import numpy as np
+    from smartredis import Client
+
+    x = np.random.rand(100, 100).astype(np.float32)
+    y = np.random.rand(100, 100).astype(np.float32) * 2 + 10
+
+    client = Client(cluster=False)
+    client.put_tensor("X_rand", x)
+    client.put_tensor("Y_rand", y)
+
+    client.set_script_from_file("shifter", "./shift.script", device="CPU")
+    client.run_script("shifter", "shift_y_to_x", inputs=["X_rand", "Y_rand"], outputs=["Y_scaled"])
+    y_scaled = client.get_tensor("Y_scaled")
+
+Simpler functions (or functions that do not require calling other user-defined
+or imported functions), can be defined inline and uploaded to the DB. For example:
+
+.. code-block:: python
+
+    import numpy as np
+    from smartredis import Client
+
+    def normalize(X):
+        mean = X.mean()
+        std = X.std()
+
+        return (X-mean)/std
+
+    x = np.random.rand(100, 100).astype(np.float32) * 2 + 10
+
+    client = Client(cluster=False)
+    client.put_tensor("X_rand", x)
+
+    client.set_function("normalizer", normalize)
+    client.run_script("normalizer", "normalize", inputs=["X_rand"], outputs=["X_norm"])
+    x_norm = client.get_tensor("X_norm")
+
+Notice that the key ``"normalizer"`` represents the script containing the function (similar to
```

Comment:
Do you mind adding more information here? I think it might be easier to understand if you provided more guidance during the 'Notice' sentence, possibly point out run_script() or specifically where the function and script are named, I might split the code up into parts as well -> *describe* set_function(), *describe* run_script()